scripts/wsrep_sst_common.sh

```diff
@@ -330,3 +330,32 @@ wsrep_check_programs()
 
     return $ret
 }
+
+
+# Returns the absolute path from a path to a file (with a filename)
+#   If a relative path is given as an argument, the absolute path
+#   is generated from the current path.
+#
+# Globals:
+#   None
+#
+# Parameters:
+#   Argument 1: path to a file
+#
+# Returns 0 if successful (path exists) and the absolute path is output.
+# Returns non-zero otherwise
+#
+function get_absolute_path()
+{
+    local path="$1"
+    local abs_path retvalue
+    local filename
+
+    filename=$(basename "${path}")
+    abs_path=$(cd "$(dirname "${path}")" && pwd)
+    retvalue=$?
+    [[ $retvalue -ne 0 ]] && return $retvalue
+
+    printf "%s/%s" "${abs_path}" "${filename}"
```

Comment:
wsrep_log_error?

Reply:
The calling function handles the error (if non-zero is returned).
The printf is outputting a value from the function by writing to stdout, it's being used like the following:
  abs_path=$(get_absolute_path 'path/to/a/file')
  if [[ $? -ne 0 ]]; then
    # handle error
  fi